carma-messenger-core/cpp_message/src/SPAT_Message.cpp

```diff
@@ -0,0 +1,702 @@
+                    for (size_t j = 0; j < state->enabledLanes->list.count; j++)
+                    {
+                        LaneID_t *enabled_lane_id = new LaneID_t;
+                        enabled_lane_id = state->enabledLanes->list.array[j];
+                        enabled_lanes_list.lane_id_list.push_back(*enabled_lane_id);
+                    }
+                    intersection.enabled_lanes = enabled_lanes_list;
+                }
+                else{
+                    
+                }
+                intersection.enabled_lanes_exists = enabled_lanes_exists;
+
+                //MovementList
+                j2735_msgs::MovementList movement_states;
+                //movement states is an array of Movement states
+                for (size_t j = 0; j < state->states.list.count; j++)
+                {
+                    if(!state->states.list.array[j]){
+                        continue;
+                    }
+                    j2735_msgs::MovementState movement_state;
+                    movement_state.movement_name_exists = false;
+                    if (state->states.list.array[j]->movementName)
+                    {
+                        movement_state.movement_name_exists = true;
+                        //movement_state.movement_name = state->states.list.array[j]->movementName;
+                        size_t len = state->states.list.array[j]->movementName->size;
+                        for (int k = 0; k < len; k++)
+                        {
+                            movement_state.movement_name += state->states.list.array[j]->movementName->buf[k];
+                        }
+
+                    }
+
+                    //Signal Group ID
+                    movement_state.signal_group = SIGNAL_GROUP_UNAVAILBALE_;
+                    if (state->states.list.array[j]->signalGroup)
+                    {
+                        movement_state.signal_group = state->states.list.array[j]->signalGroup;
+                    }
+
+                    //State Time Speed Movement Event List
+                    j2735_msgs::MovementEventList movement_event_list;
+                    for (int k = 0; k < state->states.list.array[j]->state_time_speed.list.count; k++)
+                    {
+                        if(!state->states.list.array[j]->state_time_speed.list.array[k]){
+                            continue;
+                        }
+                        j2735_msgs::MovementEvent movement_event;
+                        //movement_event.event_state = state->states.list.array[j]->state_time_speed.list.array[j]->eventState;
+                        //Decode movement event
+                        //1. MovementPhaseState
+                        if(state->states.list.array[j]->state_time_speed.list.array[k]->eventState){
+                            movement_event.event_state.movement_phase_state = state->states.list.array[j]->state_time_speed.list.array[k]->eventState;
```

Comment:
I think eventState is uin8_t, and DARK=1 for example is `01000000` which is 64 in int not 1. 
So I think we want similar decoding like you did in `//Decode status`

Reply:
For this eventState is a uint8_t not a bit array like status was. Checking the definition eventState is a MovementPhaseState which is defined as an enum ranging from 0-9
https://github.com/usdot-fhwa-stol/carma-messenger/blob/9d3a864dab2bd6bf0762ba954d1c3ed744fbc83c/carma-messenger-core/cpp_message/include/generated/MovementEvent.h#L31
https://github.com/usdot-fhwa-stol/carma-messenger/blob/develop/carma-messenger-core/cpp_message/include/generated/MovementPhaseState.h#L22

Reply:
ah I see that makes sense